thali/NextGeneration/thaliMobile.js

```diff
@@ -30,11 +30,76 @@ var getCombinedResult = function (results) {
   };
 };
 
+/**
+ * Enum to define the network types
+ *
+ * @readonly
+ * @enum {string}
+ */
+var networkTypes = {
+  WIFI: 'WIFI',
+  NATIVE: 'NATIVE',
+  BOTH: 'BOTH'
+};
+module.exports.networkTypes = networkTypes;
+
+var getWifiMethodByNetworkType = function (method, networkType) {
```

Comment:
I think we can have a good function like `getMethodByNetworkType (target, method, networkType)` where target equals `thaliWifiInfrastructure` or `ThaliMobileNativeWrapper`.